### PR TITLE
Take ownership

### PR DESCRIPTION
**Pull request by ChriMarMe**
This PR adds cmd/gosedctl as new tool which shall serve as one replacement for all others tools.
At the moment it only implements TakeOwnership command.

It is based on #88

### CODE REVIEW

File: cmd/gosedctl/cmd.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	"fmt"
+
+	"github.com/open-source-firmware/go-tcg-storage/pkg/core"
+	"github.com/open-source-firmware/go-tcg-storage/pkg/core/table"
+	"github.com/open-source-firmware/go-tcg-storage/pkg/core/uid"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+type context struct{}
```
Comment (jensdrenhaus) [inline cmd/gosedctl/cmd.go:14]: add a short comment

File: cmd/gosedctl/cmd.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+type initialSetupCmd struct {
```
Comment (jensdrenhaus) [inline cmd/gosedctl/cmd.go:16]: add a comment

File: cmd/gosedctl/cmd.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	Device   string `flag:"" required:"" short:"d"  help:"Path to SED device (e.g. /dev/nvme0)"`
+	Password string `flag:"" optional:"" short:"p"`
+	Hash     string `flag:"" optional:"" default:"sedutil-dta"`
+}
+
+var cli struct {
```
Comment (jensdrenhaus) [inline cmd/gosedctl/cmd.go:22]: Is this supposed to hold further commands?

File: cmd/gosedctl/cmd.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+func (t *initialSetupCmd) Run(ctx *context) error {
+	fmt.Printf("Open device: %s", t.Device)
+	coreObj, err := core.NewCore(t.Device)
+	if err != nil {
+		return fmt.Errorf("NewCore(%s) failed: %v", t.Device, err)
+	}
+	fmt.Println("Find ComID")
+	comID, _, err := core.FindComID(coreObj.DriveIntf, coreObj.DiskInfo.Level0Discovery)
+	if err != nil {
+		return err
```
Comment (jensdrenhaus) [inline cmd/gosedctl/cmd.go:35]: Why no fmt.Errorf() like above?

File: cmd/gosedctl/cmd.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	if err != nil {
+		return fmt.Errorf("Admin_SP_GetLifeCycleState() failed: %v", err)
+	}
+	if lcs != table.ManufacturedInactive {
+		return fmt.Errorf("LockingSP Lifecycle state of %s, but require %s", lcs.String(), table.ManufacturedInactive)
+	}
+	if err := table.LockingSPActivate(sess); err != nil {
+		return fmt.Errorf("LockingSPActivate() failed: %v", err)
+	}
+	sess.Close()
```
Comment (jensdrenhaus) [inline cmd/gosedctl/cmd.go:89]: Will be closed by defer already

File: cmd/gosedctl/cmd.go [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+	fmt.Println("Configure LockingRange0")
+	// Configure LockingRange0
+	// New Session to LockingSP required
+	sess, err = cs.NewSession(uid.LockingSP)
```
Comment (jensdrenhaus) [inline cmd/gosedctl/cmd.go:94]: Doesn't this session need to be closed?

File: cmd/gosedctl/cmd.go (hunk 5 above)
Comment (jensdrenhaus) [reply cmd/gosedctl/cmd.go:89]: Maybe split into separate functions from here on?

File: cmd/gosedctl/main.go [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"github.com/alecthomas/kong"
+)
+
+var (
```
Comment (jensdrenhaus) [inline cmd/gosedctl/main.go:7]: Can it be const?

File: pkg/core/method/method.go [hunk 8]
```diff
@@ -130,6 +130,10 @@ func (m *MethodCall) NamedBool(name string, val bool) {
 	}
 }
 
+func (m *MethodCall) Token(t stream.TokenType) {
```
Comment (jensdrenhaus) [inline pkg/core/method/method.go:133]: Add a comment

File: pkg/core/table/admin.go [hunk 9]
```diff
@@ -31,6 +32,25 @@ func Admin_C_PIN_MSID_GetPIN(s *core.Session) ([]byte, error) {
 	}
 	return pin, nil
 }
+func Admin_C_Pin_SID_SetPIN(s *core.Session, password []byte) error {
```
Comment (jensdrenhaus) [inline pkg/core/table/admin.go:35]: Add a comment.

**Review by jensdrenhaus — COMMENTED** [review]
Apart from small nits, it looks good to me codewise. Not tested nor checked with the spec yet.

File: cmd/gosedctl/cmd.go (hunk 5 above)
Comment (ChriMarMe) [reply cmd/gosedctl/cmd.go]: I renamed it to adminSession and lockingSession to separate between the sessions and they are closed at the exact point they need to be closed.

**Pull request merged**